include/buffer.h

```diff
@@ -119,11 +120,14 @@ class buffer {
 		~tracker() {
 			CELERITY_DETAIL_TRACY_ZONE_SCOPED("buffer::~buffer", DarkCyan);
 			detail::runtime::get_instance().destroy_buffer(id);
+			// The user must guarantee liveness of the user pointer only until the buffer instance goes out of scope
+			if(references_user_memory) { detail::runtime::get_instance().sync(detail::epoch_action::none); }
```

Comment:
This is more synchronization than strictly necessary. It might be possible to do this more efficiently.

Also, as discussed offline, it would be neat to have an integration test for this.

Reply:
I've added an integration test.

The fine-grained synchronization needs a refactor of fences first, so I'm postponing this.